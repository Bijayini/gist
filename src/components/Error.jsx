import React, { Fragment, Component } from 'react';

import { withLoader, loaderPropTypes } from '../lib/loader';
import Heading from '../ui-components/Heading';

import labels from './labels/Error.labels';

class Error extends Component {
  componentDidMount() {
    const {
      loader: { showLoader, updateShowLoader },
    } = this.props;
    if (showLoader) {
      updateShowLoader(false);
    }
  }

  render() {
    return (
      <Fragment>
        <div className="error">
          <Heading
            text={labels.error_message}
            color="warningMessageColor"
            variant="extra-bold"
            level="1"
          />
        </div>
        <style jsx="true">
          {`
            .error {
              text-align: center;
              margin-top: 20px;
            }
          `}
        </style>
      </Fragment>
    );
  }
}

Error.propTypes = {
  loader: loaderPropTypes.isRequired,
};

export default withLoader(Error);

export { Error };
