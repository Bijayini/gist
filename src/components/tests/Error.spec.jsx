import React from 'react';
import { shallow } from 'enzyme';

import { Error } from '../Error';

describe('<Error />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      loader: {
        showLoader: true,
        updateShowLoader: jest.fn(),
      },
    };
    component = shallow(<Error {...props} />);
  });

  it('should match with snapshot and should hide loader if loader is present', () => {
    expect(component.debug()).toMatchSnapshot();

    expect(props.loader.updateShowLoader).toHaveBeenCalledWith(false);
  });
});
