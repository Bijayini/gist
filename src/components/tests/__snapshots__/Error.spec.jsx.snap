// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should match with snapshot and should hide loader if loader is present 1`] = `
"<Fragment>
  <div className=\\"jsx-2032891786 error\\">
    <Heading text=\\"Technical Error!!\\" color=\\"warningMessageColor\\" variant=\\"extra-bold\\" level=\\"1\\" className=\\"\\" styles={false} />
  </div>
  <JSXStyle id=\\"2032891786\\">
    .error.jsx-2032891786{text-align:center;margin-top:20px;}
  </JSXStyle>
</Fragment>"
`;
